src/graph.py

```diff
@@ -0,0 +1,96 @@
+# -*- coding utf-8 -*-
+
+
+class Graph(object):
+
+    def __init__(self, initial_graph=None):
+        '''Pass in a graph in the form of a dictionary with the nodes as
+        key, and a list of edges as value, for example
+        {1:[2, 3]},
+         2:[]},
+         3:[1]}
+        '''
+        if initial_graph is None:
+            self.graph = {}
+        else:
+            self.graph = initial_graph
+
+    def __iter__(self):
+        return iter(self.graph)
+
+    def nodes(self):
+        '''return a list of all nodes in the graph'''
+        return self.graph.keys()
+
+    def edges(self):
+        '''returns a list of tuples, that are edges'''
+        return self.graph.values()
```

Comment:
This returns a list of neighbors for each node, not actual edges. An edge is a relationship between each node and its neighbors individually, e.g.:
for graph
```
g = {
    1: [2, 3, 4, 5],
    2: [4, 5],
    3: [],
    4: [6]
}
```
the printed edges should be 
```
"1-2", "1-3", "1-4", "1-5", "2-4", "2-5", "4-6"
```